WcaOnRails/app/models/competition.rb

```diff
@@ -100,16 +102,10 @@ def info
   end
 
   def build_clone
-    # Don't clone the following attributes.
-    attributes_to_clone = attributes
-    %w(id name cellName year month day endMonth endDay registration_open registration_close results_posted_at).each { |attribute| attributes_to_clone.delete attribute }
-    clone = Competition.new
-    clone.assign_attributes(attributes_to_clone)
-    clone.organizers = organizers
-    clone.delegates = delegates
-    clone.showAtAll = false
-    clone.isConfirmed = false
-    clone
+    Competition.new(attributes.except(*UNCLONEABLE_ATTRIBUTES)).tap do |clone|
```

Comment:
As you've added both attributes groups as Lars suggested, it would be better to do `attributes.slice(CLONABLE_ATTRIBUTES)` here.

This really cleaned things up.